### PR TITLE
CALC-1803 Related, Cleaned up the flow control logic

### PR DESCRIPTION
**Pull request by axeisghost**
Code coverity found out problem about flow control, mention in [comments at CALC-1803](https://issues.apache.org/jira/browse/CALCITE-1803?focusedCommentId=16089025&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16089025)

### CODE REVIEW

File: druid/src/main/java/org/apache/calcite/adapter/druid/DruidConnectionImpl.java [hunk 1]
```diff
@@ -350,24 +350,24 @@ private void parseField(List<String> fieldNames, List<ColumnMetaData.Rep> fieldT
...
           } else if (s.equals("NaN")) {
             rowBuilder.set(i, Double.NaN);
-            break;
+          } else {
+            rowBuilder.set(i, s);
           }
-          //fallthrough
+          break;
         default:
-          rowBuilder.set(i, s);
```
Comment (julianhyde) [inline druid/src/main/java/org/apache/calcite/adapter/druid/DruidConnectionImpl.java]: Why did you remove this line "rowBuilder.set(i, s);"? We still want to set the value if the type is not numeric.

Comment (axeisghost) [inline druid/src/main/java/org/apache/calcite/adapter/druid/DruidConnectionImpl.java]: Oh, It is totally my bad. I haven't touched this part for a while, so I kinda forget the original purpose of default branch.

**Comment by axeisghost** [comment]
added the line back

File: druid/src/main/java/org/apache/calcite/adapter/druid/DruidConnectionImpl.java (hunk 1 above)
Comment (julianhyde) [inline druid/src/main/java/org/apache/calcite/adapter/druid/DruidConnectionImpl.java]: I modified your code so that exceptional conditions throw or return, and non-exceptional conditions all go through the same 'rowBuilder.set(i, s)'. See https://github.com/julianhyde/calcite/commit/cbbe627feac1293796f68ff8ec8d4bd7b64724d2, and let me know if there's a problem. Otherwise I'll commit shortly.

**Comment by axeisghost** [comment]
Thank you. It is good. Thanks for helping me fix the logic.
